py/repl.c

```diff
@@ -187,6 +187,11 @@ size_t mp_repl_autocomplete(const char *str, size_t len, const mp_print_t *print
                 if (s_len <= d_len && strncmp(s_start, d_str, s_len) == 0) {
                     mp_load_method_protected(obj, q, dest, true);
                     if (dest[0] != MP_OBJ_NULL) {
+                        // special case; filter out words that begin with underscore
+                        // unless there's already a partial match
+                        if (s_len == 0 && *d_str == '_') {
```

Comment:
nitpick: would ``d_str[0]`` be more readable here?

Reply:
I haven't programmed C in years lol, forgot that was an option. I like making the first character check explicit; will update soon